src/app/modules/shared_models/token_permissions_model.nim

```diff
@@ -145,11 +145,9 @@ QtObject:
     self.items[idx].tokenCriteriaMet = item.tokenCriteriaMet
 
     let index = self.createIndex(idx, 0, nil)
+    defer: index.delete
     self.dataChanged(index, index, @[
-      ModelRole.Id.int,
-      ModelRole.Key.int,
       ModelRole.Type.int,
-      ModelRole.TokenCriteria.int,
```

Comment:
Why `TokenCriteria` is omitted here? It's changed above. Moreover `chatList` is changed but not present in `dataChanged` call.

Reply:
It's a (sub)model but yeah, you are probably right, I'll revert that part and also add `chatList`

Reply:
Fixed